cogs/roles.py

```diff
@@ -0,0 +1,163 @@
+from discord import utils
+from discord.ext import commands
+
+import codecs
+import configparser
+from .utils.paginator import Pages
+
+
+class Roles(commands.Cog):
+    def __init__(self, bot):
+        self.bot = bot
+        self.roles = self.bot.config.roles
+
+    async def toggleRole(self, ctx, transaction, requestedRole, category):
+        """
+        Assigs a single role to a user with no checks from a category of roles
+        """
+        member = ctx.message.author
+        roles = self.roles.get(category).split(", ")
+
+        # normalize user input to title case
+        requestedRole = requestedRole.title()
+
+        if (transaction == "add") and (requestedRole in roles):
+            role = discord.utils.get(ctx.guild.roles, name=requestedRole)
+            await member.add_roles(role, reason="Self Requested")
+        elif (transaction == "remove") and (requestedRole in roles):
+            role = discord.utils.get(ctx.guild.roles, name=requestedRole)
+            await member.remove_roles(role, reason="Self Requested")
+        else:
+            await ctx.send("Error, that role doesn't exist in this category")
+        return
+
+    async def toggleRestrictedRole(self, ctx, transaction, requestedRole,
+                                   category):
+        """
+        Assigs a single role to a user with no checks from a category of roles
+        """
+        member = ctx.message.author
+        roles = self.roles.get(category).split(", ")
+
+        # normalize user input to title case
+        requestedRole = requestedRole.title()
+
+        if (transaction == "add") and (requestedRole in roles):
+            role = discord.utils.get(ctx.guild.roles, name=requestedRole)
+            # Allow only a single faculty role per user.
+            for category_roles in roles:
+                old_role = discord.utils.get(ctx.guild.roles,
+                                             name=category_roles)
+                if old_role is not None:
+                    await member.remove_roles(old_role,
+                                              reason="Self Requested")
+            await member.add_roles(role, reason="Self Requested")
+        elif (transaction == "remove") and (requestedRole in roles):
+            role = discord.utils.get(ctx.guild.roles, name=requestedRole)
+            await member.remove_roles(role, reason="Self Requested")
+        else:
+            await ctx.send("Error, that role doesn't exist in this category")
+        return
+
+    @commands.command()
+    async def pronoun(self, ctx, transaction, pronoun):
+        """
+        Self-assign a pronoun role to a user. 
+        """
+        await Roles.toggleRole(self, ctx, transaction, pronoun, "pronouns")
+
+    @commands.command()
+    async def field(self, ctx, transaction, field):
+        """
+        Self-assign a field of study role to a user. 
+        """
+        await Roles.toggleRole(self, ctx, transaction, field, "fields")
```

Comment:
not a static fn, call with `self.toggleRole` and skip passing the self arg by hand
same goes for all of these calls